.circleci/config.yml

```diff
@@ -38,67 +51,127 @@ commands:
+          name: Copy config.json from the last release
+      - run: yarn sign-apv
+      - run: 
+          command: |
+            if [[ "$CIRCLE_BRANCH" == "main" ]]
+            then
               yarn release
-            } else {
+            else
               yarn run pack
-            }
-      - run: 
-          command: 7z a -r ../../pack-dist/Windows.zip *
-          working_directory: pack/Nine Chronicles-win32-x64/
+            fi
+      - when:
+          condition: 
+            equal: [ *windows, << parameters.os >>]
+          steps:
+          - run: 
+              command: 7z a -r ../../pack-dist/Windows.zip *
+              working_directory: pack/Nine Chronicles-win32-x64/
+      - when:
+          condition:
+            equal: [ *macos, << parameters.os >>]
+          steps:
+            - run:
+                command: |
+                  chmod +x "Nine Chronicles.app/Contents/Resources/app/9c.app/Contents/MacOS/9c"
+                  tar cvfz ../../pack-dist/macOS.tar.gz *
+                working_directory: pack/Nine Chronicles-darwin-x64/
+      - when:
+          condition:
+            equal: [ *linux, << parameters.os >>]
+          steps:
+            - run: 
+                command: |
+                  chmod +x resources/app/9c
+                  tar cvfz ../../pack-dist/Linux.tar.gz *
+                working_directory: pack/Nine Chronicles-linux-x64/
       - store_artifacts:
           path: pack-dist
           destination: dist-artifacts
-      - run:
+      - run: 
           command: |
-            Invoke-WebRequest https://awscli.amazonaws.com/AWSCLIV2.msi -OutFile AWSCLIV2.msi
-            msiexec.exe /i AWSCLIV2.msi /quiet
-            [Environment]::SetEnvironmentVariable("Path", $env:PATH + ";c:\Program Files\Amazon\AWSCLIV2", "Machine")
-
-            aws configure set aws_access_key_id $env:AWS_ACCESS_KEY
-            aws configure set aws_secret_access_key $env:AWS_SECRET_KEY
-
-            aws s3 cp pack-dist/Windows.zip s3://9c-artifacts/9c-launcher/$env:CIRCLE_SHA1/Windows.zip
-          name: Upload to S3
+            aws configure set aws_access_key_id $AWS_ACCESS_KEY
+            aws configure set aws_secret_access_key $AWS_SECRET_KEY
+          name: Set S3
+      - when:
+          condition: 
+            equal: [ *windows, << parameters.os >>]
+          steps:
+            - run: 
+                command: aws s3 cp pack-dist/Windows.zip s3://9c-artifacts/9c-launcher/$CIRCLE_SHA1/Windows.zip
+      - when:
+          condition:
+            equal: [ *macos, << parameters.os >>]
+          steps:
+            - run: 
+                command: aws s3 cp pack-dist/macOS.tar.gz s3://9c-artifacts/9c-launcher/$CIRCLE_SHA1/macOS.tar.gz
+      - when:
+          condition:
+            equal: [ *linux, << parameters.os >>]
+          steps:
+            - run: 
+                command: aws s3 cp pack-dist/Linux.tar.gz s3://9c-artifacts/9c-launcher/$CIRCLE_SHA1/Linux.tar.gz
```

Comment:
Could we simplify this by any chance?

Reply:
[Sadly there's no `else if` in circleci's logic statements](https://circleci.com/docs/configuration-reference#logic-statements), so I'm pretty sure this is best we can do for now.